### ISSUE TITLE
[Accessibility]: Code Insights — Share link modal pronounces full URL string

### ISSUE THREAD

**Issue by vovakulikov** [sourcegraph/sourcegraph#41815]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33502

### Problem description

<img width="559" alt="Screenshot 2022-09-20 at 20 33 52" src="https://user-images.githubusercontent.com/18492575/191325862-b8effa72-c846-49f4-9ee2-bc72fce8d374.png">

Screen reader announces that modal has 4 items (which is confused). Screen reader doesn't announce anything about modal type (which is sharable link wizard). URL input has autofocus so screen readers skips all important content above input in the sharable link modal. 

### Expected behavior

Screen readers announce 
- Type of the modal (shareable link wizard)
- Reads modal description (short version, that shareable link is available or not) 
- Announces that URL input has autofocus (and read the short version of link? not sure)
- After link is copied and modal window is closed screen reader should announce that modal has been closed.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
